src/uwtools/tests/utils/test_file.py

```diff
@@ -24,6 +25,35 @@ def assets(tmp_path):
     return now, renamed, rundir
 
 
+def test_StdinProxy():
+    msg = "proxying stdin"
+    with patch.object(sys, "stdin", new=StringIO(msg)):
+        assert sys.stdin.read() == msg
+        # Reading from stdin a second time yields no input, as the stream has been exhausted:
+        assert sys.stdin.read() == ""
+    with patch.object(sys, "stdin", new=StringIO(msg)):
+        sp = file.StdinProxy()
+        assert sp.read() == msg
+        # But the stdin proxy can be read multiple times:
+        assert sp.read() == msg
+
+
+def test__stdinproxy():
+    file._stdinproxy.cache_clear()
+    msg0 = "hello world"
+    msg1 = "bonjour monde"
+    # Unsurprisingly, the first read from stdin finds the expected message:
+    with patch.object(sys, "stdin", new=StringIO(msg0)):
+        assert file._stdinproxy().read() == msg0
+    # But after re-patching stdin with a new message, a second read returns the old message:
+    with patch.object(sys, "stdin", new=StringIO(msg1)):
+        assert file._stdinproxy().read() == msg0  # <-- the OLD message
+    # However, if the cache is cleared, the second message is then read:
+    file._stdinproxy.cache_clear()
+    with patch.object(sys, "stdin", new=StringIO(msg1)):
+        assert file._stdinproxy().read() == msg1  # <-- the NEW message
```

Comment:
Super interesting use, comments were definitely useful here to understand what was happening in these lines of code.